server/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java

```diff
@@ -43,6 +46,55 @@ public void setDocument(int docId) {
         this.docId = docId;
     }
 
+    private static class FormatKey {
+        final String field;
+        final DocValueFormat format;
+
+        private FormatKey(String field, DocValueFormat format) {
+            this.field = field;
+            this.format = format;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            FormatKey formatKey = (FormatKey) o;
+            return Objects.equals(field, formatKey.field) && Objects.equals(format, formatKey.format);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(field, format);
+        }
+    }
+
+    public ScriptDocValues<?> get(String name, DocValueFormat format) {
```

Comment:
This is largely analogous to `get(field)` below, with the addition of formatting.  The FormatKey is so that a `fields` call can request the same field multiple times with different formats.